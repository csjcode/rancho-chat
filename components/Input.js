import { StyleSheet, Text, TextInput, View } from 'react-native'

import colors from '../constants/colors/colors'
import getColors from '../constants/colors/getColors'
import { useState } from 'react'

const colorsTheme = getColors()

const Input = (props) => {
  // console.log(`Input props ${JSON.stringify(props)}`)
  var colorsTheme = getColors()
  const [value, setValue] = useState(props.initialValue)

  const onChangeText = (text) => {
    console.log(text)
    setValue(text)
    props.onInputChanged(props.id, text)
  }

  const styleMerge = (className) => {
    return {
      ...stylesFor(colorsTheme)[className],
      ...props?.addStyles?.[className],
    }
  }

  return (
    <View style={styleMerge('container')}>
      <Text style={styleMerge('label')}>{props.label}</Text>
      <View style={styleMerge('inputContainer')}>
        {props.icon && (
          <props.iconPack
            name={props.icon}
            size={props.iconSize || 15}
            style={stylesFor(colorsTheme).input}
          />
        )}
        <TextInput
          {...props}
          style={styleMerge('inputText')}
          onChangeText={onChangeText}
          value={value}
        />
      </View>

      {props.errorText && (
        <View style={stylesFor(colorsTheme).errorContainer}>
          <Text style={stylesFor(colorsTheme).errorText}>
            {props.errorText[0]}
          </Text>
        </View>
      )}
    </View>
  )
}

const stylesFor = (colorsTheme) =>
  StyleSheet.create({
    container: {
      width: '100%',
      // marginBottom: 30,
    },
    label: {
      marginVertical: 8,
      fontFamily: 'bold',
      letterSpacing: 0.3,
      color: colorsTheme.textColor,
    },
    inputContainer: {
      width: '100%',
      // backgroundColor: '#abd5c6',
      paddingHorizontal: 10,
      paddingVertical: 15,
      borderRadius: 2,
      backgroundColor: colorsTheme.formInputBackgroundColor,
      flexDirection: 'row',
      alignItems: 'center',
    },
    icon: {
      marginRight: 10,
      color: colorsTheme.grey,
    },
    input: {
      color: colorsTheme.formInputTextColor,
      fontFamily: 'regular',
      letterSpacing: 0.3,
      paddingTop: 0,
    },
    inputText: {
      paddingLeft: 10,
      paddingRight: 10,
      width: '100%',
      color: colorsTheme.formInputTextColor,
    },
    errorContainer: {
      marginVertical: 5,
    },
    errorText: {
      color: 'red',
      fontSize: 13,
      fontFamily: 'regular',
      letterSpacing: 0.3,
    },
  })

export default Input
