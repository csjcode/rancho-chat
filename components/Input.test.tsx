import React from "react";
import renderer from "react-test-renderer";
import Input from "./Input";

test("Input component renders correctly", () => {
  const tree = renderer
    .create(
      <Input
        initialValue="Test value"
        label="Test label"
        icon="testIcon"
        iconPack="testIconPack"
        iconSize={15}
        addStyles={{ testStyle: "testValue" }}
        onInputChanged={() => {}}
        errorText={["Test error"]}
        id={1}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
