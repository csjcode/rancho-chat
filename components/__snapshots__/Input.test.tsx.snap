// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input component renders correctly 1`] = `
<View
  style={
    Object {
      "width": "100%",
    }
  }
>
  <Text
    style={
      Object {
        "color": "#32d48e",
        "fontFamily": "bold",
        "letterSpacing": 0.3,
        "marginVertical": 8,
      }
    }
  >
    Test label
  </Text>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#abd5c6",
        "borderRadius": 2,
        "flexDirection": "row",
        "paddingHorizontal": 10,
        "paddingVertical": 15,
        "width": "100%",
      }
    }
  >
    <testIconPack
      name="testIcon"
      size={15}
      style={
        Object {
          "color": "#000",
          "fontFamily": "regular",
          "letterSpacing": 0.3,
          "paddingTop": 0,
        }
      }
    />
    <TextInput
      addStyles={
        Object {
          "testStyle": "testValue",
        }
      }
      errorText={
        Array [
          "Test error",
        ]
      }
      icon="testIcon"
      iconPack="testIconPack"
      iconSize={15}
      id={1}
      initialValue="Test value"
      label="Test label"
      onChangeText={[Function]}
      onInputChanged={[Function]}
      style={
        Object {
          "color": "#000",
          "paddingLeft": 10,
          "paddingRight": 10,
          "width": "100%",
        }
      }
      value="Test value"
    />
  </View>
  <View
    style={
      Object {
        "marginVertical": 5,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "red",
          "fontFamily": "regular",
          "fontSize": 13,
          "letterSpacing": 0.3,
        }
      }
    >
      Test error
    </Text>
  </View>
</View>
`;
